### PR TITLE
feat(docz): ordering menu on doczrc

### PR DESCRIPTION
**Pull request by mpivaa**
### Ordering menu on doczrc

Based on http://feedback.docz.site/roadmap/p/managing-menu-order-on-doczrc

### Pre-merge checklist

- [x] Order menu through doczrc
- [ ] Update docs

### CODE REVIEW

**Comment by pedronauck** [comment]
Great job @mpivaa, you're doing a really great work, thanks 👏 

So, about this task, sorry to tell you just now, but I was thinking of something a little bit different and maybe a little bit simple too...

The idea is to use the `menu` property` to create the hierarchy and be able to add some custom properties for each one:

```js
export default {
  menu: [
    'Getting Started',
    {
      name: 'Components',
      menu: [
        'Alerts',
        'Button',
        ...
      ],
    },
    {
      name: 'Github',
      route: 'https://github.com/pedronauck/docz'
    }
  ]
}
```

But this is kinda a breaking change because of today when we use `<Docs>` component inside a theme to get the array of menus, this array just has strings. And to keep maintainability, we should need to use the `menu` to ordering and populate the data.

File: packages/docz/src/components/Docs.tsx [hunk 1]
```diff
@@ -8,12 +8,60 @@ import { configSelector } from './ThemeConfig'
...
 export const isFn = (value: any): boolean => typeof value === 'function'
 
-const sortBy = (a: any, b: any, reverse?: boolean) => {
+const compare = (a: any, b: any, reverse?: boolean) => {
   if (a < b) return reverse ? 1 : -1
   if (a > b) return reverse ? -1 : 1
   return 0
 }
 
+const UNKNOWN_POS = 999999  // TODO: fix this?
```
Comment (pedronauck) [inline packages/docz/src/components/Docs.tsx]: I was thinking to try something like that:
https://github.com/mdx-js/mdx/blob/master/docs/_app.js#L49-L60

What do you think about @mpivaa ?

Comment (mpivaa) [reply packages/docz/src/components/Docs.tsx]: I thought about `Infinity` too, I will change

**Comment by mpivaa** [comment]
I think these changes makes sense, but definitely they are breaking changes, maybe we should always pass an object as the `menu` and the theme handles how to present it?

**Comment by pedronauck** [comment]
Yeah, for sure @mpivaa, this will be a breaking change definitely 😕 So, I think that is best we keep baby steps here and release this feature like people want, in a simple way and. Maybe leave this change to release in v1.0 and we can put some others nice things together ✌️

**Comment by hnordt** [comment]
@pedronauck I can see you are not using `menu` in `doczrc.js`. Why you don't allow both passing `menu` and keep the current behavior when it's omitted. It would prevent a breaking change.

**Comment by mpivaa** [comment]
@hnordt It would break if the user defines `menu` in `doczrc.js` but the theme is not prepared for an object.

**Comment by mpivaa** [comment]
@pedronauck I've changed the config interface so it looks like:

```js
export default {
  menu: [
    'Getting Started',
    {
      name: 'Components',
      docs: [
        'Alerts',
        'Button',
        ...
      ],
    }
  ]
}
```

That way when we change the implementation of `Menu` to allow objects, it will not break old configurations.

**Review by pedronauck — APPROVED** [review]

**Comment by alexlee-dev** [comment]
If I were to try and implement this now, would it work? v0.11.2

**Comment by sidewalksalsa** [comment]
What version will this be released on @pedronauck @mpivaa ?

**Comment by pedronauck** [comment]
Will be released in the `v0.12.0` @sidewalksalsa, I'm have just some bugs and performance improvements to solve before release this version 🙏

**Pull request merged**